### DISCUSSION TITLE
Add option to use prism.js syntax highlighting, instead of highlight.js

### DISCUSSION THREAD

**Discussion by ftessier** [hakimel/reveal.js#3514] [Ideas]
The ability to highlight syntax automatically in code frames is wonderful! It would be even better if one could select alternative highlighting libraries, such as `prism.js` which covers more languages. For example, I am working on a git presentation, and `highlight.js` does not highlight syntax for git commands.

**Comment by hakimel** [comment]
We're deeply tied into highlight.js for things like auto-animated code and line highlights/steps so we can't offer a choice of libraries. If highlight.js falls behind and there superior alternatives out there we could consider a full switch in the future though.

**Comment by t-fritsch** [comment]
on this topic, Highlight.js team launched a survey in which they ask for feedback on Highlight vs Prism here : https://github.com/highlightjs/highlight.js/issues/3625

This has been cross-posted in Prism repo : https://github.com/PrismJS/prism/discussions/3557

Regarding adoption and weekly download, they're quite in the same range (but the trends differ) : https://npmtrends.com/highlight.js-vs-prismjs
![image](https://github.com/hakimel/reveal.js/assets/10920280/6b8e2ed2-9a79-4b71-97f2-6b0cfd69e635)


Prism v1 is not updated since more than a year, but they're working on a v2 ([last commit in june](https://github.com/PrismJS/prism/commits/v2)) : https://github.com/PrismJS/prism/discussions/3531

I remember having used prism on old reveal version but haven't been able to reproduce it with v4+

**Reply by ftessier** [reply]
Thanks for considering and for the insight, understood!
